extension/chrome/settings/setup/setup-key-manager-autogen.ts

```diff
@@ -19,20 +18,21 @@ export class SetupWithEmailKeyManagerModule {
   constructor(private view: SetupView) {
   }
 
-  public setupWithEkmThenRenderSetupDone = async () => {
-    if (!this.view.orgRules.mustAutogenPassPhraseQuietly()) {
-      const notSupportedErr = 'Combination of org rules not yet supported: PRV_AUTOIMPORT_OR_AUTOGEN cannot yet be used without PASS_PHRASE_QUIET_AUTOGEN.';
-      await Ui.modal.error(`${notSupportedErr}\n\nPlease write human@flowcrypt.com to add support.`);
-      window.location.href = Url.create('index.htm', { acctEmail: this.view.acctEmail });
-      return;
-    } else if (this.view.orgRules.forbidStoringPassPhrase()) {
-      const notSupportedErr = 'Combination of org rules not valid: PASS_PHRASE_QUIET_AUTOGEN cannot be used together with FORBID_STORING_PASS_PHRASE.';
-      await Ui.modal.error(notSupportedErr);
-      window.location.href = Url.create('index.htm', { acctEmail: this.view.acctEmail });
+  public continueEkmSetupHandler = async () => {
+    if (! await this.view.isCreatePrivateFormInputCorrect('step_2_ekm_choose_pass_phrase')) {
       return;
     }
-    const passphrase = PgpPwd.random(); // mustAutogenPassPhraseQuietly
-    const setupOptions: SetupOptions = { passphrase_save: true, submit_main: this.view.orgRules.canSubmitPubToAttester(), submit_all: false, passphrase };
+    const passphrase = $('#step_2_ekm_choose_pass_phrase .input_password').val();
+    await this.setupWithEkmThenRenderSetupDone(typeof passphrase === 'string' ? passphrase : '');
+  }
+
+  public setupWithEkmThenRenderSetupDone = async (passphrase: string) => {
+    const setupOptions: SetupOptions = {
+      passphrase_save: this.view.orgRules.mustAutogenPassPhraseQuietly() || Boolean($('#step_2_ekm_choose_pass_phrase .input_passphrase_save').prop('checked')),
```

Comment:
Do we need a test to untick "Save Pass phrase" and check that the passphrase doesn't get in to the local storage?

Reply:
When we are doing setup with EKM, I would not give the user any choice regarding pass phrase storage - instead we will infer from OrgRules. If pass phrase in memory is required then we use that, else we put it in storage without asking the user.

(big orgs usually want a consistent setup for everybody)

To anwer your question - that means there is no checkbox to uncheck.

Reply:
Actually, since you already did this, it won't hurt to leave it as is.

At least it's more explicit.

I don't think it needs a distinct test where user unchecks it, looks ok to me.